### PR TITLE
feat(relay): drastically improve usefulness of spans

### PR DESCRIPTION
**Pull request by thomaseizinger**

### CODE REVIEW

**Comment by AndrewDryga** [comment]
@thomaseizinger can you please fix the conflicts?

**Comment by thomaseizinger** [comment]
> @thomaseizinger can you please fix the conflicts?

Done!

File: rust/relay/src/main.rs [hunk 1]
```diff
@@ -105,60 +106,15 @@ async fn main() -> Result<()> {
...
-                    tracing::info!("Received init message from portal on topic {topic}, starting relay activities");
-                    break Some(channel);
-                }
-                other => {
-                    tracing::debug!("Unhandled message from portal: {other:?}");
-                }
-            }
-        }
+        connect_to_portal(&args, token, url, stamp_secret)
+            .instrument(span)
```
Comment (conectado) [inline rust/relay/src/main.rs:116]: Why not use the `intrument` attribute?

File: rust/relay/src/main.rs [hunk 2]
```diff
@@ -362,13 +379,19 @@ where
     }
 
     fn poll(&mut self, cx: &mut std::task::Context<'_>) -> Poll<Result<()>> {
+        let span = tracing::error_span!("Eventloop::poll");
```
Comment (conectado) [inline rust/relay/src/main.rs:382]: If you instrument this future this span is automatically created

File: rust/relay/src/server.rs [hunk 3]
```diff
@@ -313,7 +314,7 @@ where
         sender: SocketAddr,
         allocation_id: AllocationId,
     ) {
-        let span = tracing::error_span!("peer", %sender, %allocation_id, recipient = field::Empty, channel = field::Empty);
+        let span = tracing::error_span!("handle_relay_input", %sender, %allocation_id, recipient = field::Empty, channel = field::Empty);
```
Comment (conectado) [inline rust/relay/src/server.rs:317]: With this span (and the others) we could use the `instrument` macro, I feel it is less noisy, unless there's another reason to enter these explicitly.

**Comment by thomaseizinger** [comment]
Thanks for the review @conectado !

Mind if I address them in a follow-up PR? I'd like to see them on Google Cloud first. So far I've only looked at them in a local Jaeger instance.

**Review by conectado — APPROVED** [review]

**Pull request merged**